pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -147,8 +156,32 @@ func (c *Controller) logPacket(pktIn *ofctrl.PacketIn) error {
 		return fmt.Errorf("received error while handling packetin for NetworkPolicy: %v", err)
 	}
 
-	// Store log file
-	AntreaPolicyLogger.Printf("%s %s %s %s SRC: %s DEST: %s %d %s", ob.tableName, ob.npRef, ob.disposition, ob.ofPriority, ob.srcIP, ob.destIP, ob.pktLength, ob.protocolStr)
+	// Create 1 sec buffers for deduplicating non-Allow packet log.
+	logMsg := fmt.Sprintf("%s %s %s %s SRC: %s DEST: %s %d %s", ob.tableName, ob.npRef, ob.disposition, ob.ofPriority, ob.srcIP, ob.destIP, ob.pktLength, ob.protocolStr)
+	if ob.disposition == openflow.DispositionToString[openflow.DispositionAllow] {
+		AntreaPolicyLogger.Printf(logMsg)
+	} else {
+		_, ok := logDeduplication[logMsg]
+		// Increase count and renew buffer if duplicated within 1 sec, create buffer otherwise.
+		if ok {
+			logDeduplication[logMsg].count++
+			logDeduplication[logMsg].bufferTimer.Reset(time.Second)
+		} else {
+			record := logDedup{1, time.Now(), time.NewTimer(time.Second)}
+			logDeduplication[logMsg] = &record
+
+			// Go routine for logging when buffer timer stops.
+			go func(logKey string) {
+				<-logDeduplication[logKey].bufferTimer.C
+				if logDeduplication[logKey].count == 1 {
```

Comment:
looks like there is a possible race condition here: we access the map and its values from 2 different goroutines

Reply:
Thanks for reminder!